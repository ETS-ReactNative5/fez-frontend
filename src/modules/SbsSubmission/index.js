export { default as SbsSubmission } from './containers/SbsSubmission';
