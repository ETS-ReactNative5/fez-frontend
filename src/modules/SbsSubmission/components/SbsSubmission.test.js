import SbsSubmission from './SbsSubmission';
import Immutable from 'immutable';
import { default as formLocale } from 'locale/publicationForm';

function setup(testProps, isShallow = true) {
    const props = {
        'array': {
            insert: jest.fn(),
            move: jest.fn(),
            pop: jest.fn(),
            push: jest.fn(),
            remove: jest.fn(),
            removeAll: jest.fn(),
            shift: jest.fn(),
            splice: jest.fn(),
            swap: jest.fn(),
            unshift: jest.fn(),
        },
        autofill: jest.fn(),
        blur: jest.fn(),
        change: jest.fn(),
        clearAsyncError: jest.fn(),
        'anyTouched': true,
        'asyncValidating': false,
        asyncValidate: jest.fn(),
        clearFields: jest.fn(),
        clearSubmitErrors: jest.fn(),
        destroy: jest.fn(),
        dispatch: jest.fn(),
        handleSubmit: jest.fn(),
        initialize: jest.fn(),
        reset: jest.fn(),
        resetSection: jest.fn(),
        touch: jest.fn(),
        submit: jest.fn(),
        untouch: jest.fn(),
        clearSubmit: jest.fn(),
        'dirty': true,
        'form': 'form',
        'initialized': false,
        'submitFailed': false,
        'valid': true,
        pure: true,
        // common immutable props above
        formValues: testProps.initialValues ? Immutable.Map(testProps.initialValues) : Immutable.Map({}),
        submitting: testProps.submitting || false, // : PropTypes.bool
        submitSucceeded: testProps.submitSucceeded || false, // : PropTypes.bool
        invalid: testProps.invalid || false, // : PropTypes.bool
        pristine: testProps.pristine || false, // : PropTypes.bool
        isHdrThesis: testProps.isHdrThesis || false, // : PropTypes.bool
        fileAccessId: testProps.fileAccessId || 3, // PropTypes.number
        actions: {
            logout: jest.fn(),
            checkSession: jest.fn(),
            clearSessionExpiredFlag: jest.fn(),
        },
        ...testProps,
    };

    return getElement(SbsSubmission, props, isShallow);
}

describe('SbsSubmission test', () => {
    it('should render sbs thesis submission form', () => {
        const wrapper = setup({ isHdrThesis: false });
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('Field').length).toEqual(12);
        expect(wrapper.find('WithStyles(Button)').length).toEqual(2);
    });

    it('should render sbs thesis submission form', () => {
        const wrapper = setup({ isHdrThesis: false });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render sbs thesis submission acknowledgement', () => {
        const wrapper = setup({ isHdrThesis: false, submitSucceeded: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with all fields disabled', () => {
        const wrapper = setup({ submitting: true });
        wrapper.find('Field').forEach(field => {
            expect(field.props().disabled).toEqual(true);
        });
    });

    it('should disable submit button if invalid form data before submit', () => {
        const wrapper = setup({ disableSubmit: true });
        expect(wrapper.find('WithStyles(Button)').length).toEqual(2);

        wrapper.find('WithStyles(Button)').forEach(field => {
            if (field.props().label == formLocale.thesisSubmission.submit) {
                expect(field.props().disabled).toEqual(true);
            }
        });
    });

    it('should not disable submit button if form submit has failed', () => {
        const wrapper = setup({ submitFailed: true });
        expect(wrapper.find('WithStyles(Button)').length).toEqual(2);

        wrapper.find('WithStyles(Button)').forEach(field => {
            if (field.props().label == formLocale.thesisSubmission.submit) {
                expect(field.props().disabled).toEqual(false);
            }
        });
    });

    it('should ask when redirecting from form with data (even if submit failed)', () => {
        const wrapper = setup({ dirty: true, submitSucceeded: false });
        expect(wrapper.find('NavigationDialogBox').length).toEqual(1);
    });

    it('should not ask when redirecting from form with data after successful submit', () => {
        const wrapper = setup({ dirty: true, submitSucceeded: true });
        expect(wrapper.find('NavigationDialogBox').length).toEqual(0);
    });

    it('should display successfull submission screen', () => {
        const wrapper = setup({});
        wrapper.setProps({ submitSucceeded: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should redirect to after submit page', () => {
        window.location.assign = jest.fn();
        const wrapper = setup({}).instance()
            .afterSubmit();
        expect(window.location.assign).toBeCalledWith(expect.stringContaining(formLocale.thesisSubmission.afterSubmitLink));
    });

    it('should display confirmation box before submission', () => {
        const testMethod = jest.fn();
        const wrapper = setup({});
        wrapper.instance().depositConfirmationBox = { showConfirmation: testMethod };
        wrapper.instance().openDepositConfirmation();
        expect(testMethod).toHaveBeenCalled();
    });

    it('should display confirmation box after submission', () => {
        const testMethod = jest.fn();
        const wrapper = setup({});
        wrapper.instance().openDepositConfirmation = testMethod;
        wrapper.setProps({ isSessionValid: true, submitting: false });
        wrapper.update();
        expect(testMethod).toHaveBeenCalled();
    });

    it('deposit() method', () => {
        const testMethod = jest.fn();
        const wrapper = setup({ actions: { checkSession: testMethod } });
        wrapper.instance().deposit();
        expect(testMethod).toHaveBeenCalled();
    });

    it('cancelSubmit() method', () => {
        const testMethod = jest.fn();
        const wrapper = setup({});
        delete global.window.location;
        global.window.location = { reload: testMethod };
        wrapper.instance().cancelSubmit();
        expect(testMethod).toHaveBeenCalled();
    });

    it('setDepositConfirmation(ref) method', () => {
        const ref = 'Hello';
        const wrapper = setup({});
        wrapper.instance().setDepositConfirmation(ref);
        expect(wrapper.instance().depositConfirmationBox).toEqual(ref);
    });

    it('should reload when told to', () => {
        const wrapper = setup({ initialValues: {} });
        wrapper.instance().afterFailedSubmit();
    });

    it('should render sbs thesis submission form', () => {
        const wrapper = setup({ newRecordFileUploadingOrIssueError: true, submitSucceeded: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
