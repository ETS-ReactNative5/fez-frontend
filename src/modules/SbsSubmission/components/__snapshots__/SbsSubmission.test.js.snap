// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbsSubmission test should display successfull submission screen 1`] = `
<StandardPage
  title="Professional doctorate deposit"
>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <StandardCard
        title="Your thesis has been deposited"
      >
        <WithStyles(Typography)>
          You will receive an email confirming your thesis deposit shortly.
        </WithStyles(Typography)>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={true}
    />
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Button)
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Return to the Graduate School website
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`SbsSubmission test should render sbs thesis submission acknowledgement 1`] = `
<StandardPage
  title="Professional doctorate deposit"
>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <StandardCard
        title="Your thesis has been deposited"
      >
        <WithStyles(Typography)>
          You will receive an email confirming your thesis deposit shortly.
        </WithStyles(Typography)>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={true}
    />
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Button)
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Return to the Graduate School website
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`SbsSubmission test should render sbs thesis submission form 1`] = `
<StandardPage
  title="Professional doctorate deposit"
>
  Hello
  <form>
    <NavigationDialogBox
      txt={
        Object {
          "cancelButtonLabel": "No",
          "confirmButtonLabel": "Yes",
          "confirmationMessage": "Are you sure you want to abandon workflow?",
          "confirmationTitle": "Abandon workflow",
        }
      }
      when={true}
    />
    <WithStyles(ConfirmDialogBox)
      locale={
        Object {
          "cancelButtonLabel": "No, continue editing",
          "confirmButtonLabel": "Yes, deposit thesis",
          "confirmationMessage": "You are about to deposit your thesis with attached files. Are you sure you want to proceed?",
          "confirmationTitle": "Thesis deposit",
        }
      }
      onAction={[MockFunction]}
      onRef={[Function]}
    />
    <WithStyles(Grid)
      container={true}
      spacing={24}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Thesis information"
        >
          <WithStyles(Grid)
            container={true}
            spacing={24}
          >
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Typography)
                variant="caption"
              >
                Thesis title
              </WithStyles(Typography)>
              <Field
                component={[Function]}
                disabled={false}
                height={50}
                name="thesisTitle"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                fullWidth={true}
                label="Author name"
                name="currentAuthor.0.nameAsPublished"
                placeholder=""
                required={true}
                rows={1}
                type="text"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                itemsList={
                  Array [
                    "MPhil Thesis",
                    "PhD Thesis",
                    "Professional Doctorate",
                  ]
                }
                locale={
                  Object {
                    "label": "Thesis type",
                  }
                }
                name="rek_genre_type"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                floatingLabelText="Institution name"
                hintText=""
                name="fez_record_search_key_org_name.rek_org_name"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                floatingLabelText="Enrolling unit"
                hintText="Enrolling unit, eg. School of Business"
                name="fez_record_search_key_org_unit_name.rek_org_unit_name"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                allowPartial={true}
                className="requiredHintField"
                component={[Function]}
                disabled={false}
                floatingTitle="Publication date"
                floatingTitleRequired={true}
                name="rek_date"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Typography)
                variant="caption"
              >
                Abstract
              </WithStyles(Typography)>
              <Field
                component={[Function]}
                disabled={false}
                name="thesisAbstract"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Supervisors"
        >
          <Field
            component={[Function]}
            disabled={false}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "addButton": "Add supervisor",
                    "descriptionStep1": "",
                    "descriptionStep1NoStep2": "List your current supervisors",
                    "identifierLabel": "UQ identifier (if available)",
                    "nameAsPublishedHint": "",
                    "nameAsPublishedLabel": "Enter each supervisorâ€™s name on a separate line e.g. Firstname Surname",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "contributorAssignmentColumn": "Select your name",
                    "deleteAll": "Remove all items",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all items?",
                      "confirmationTitle": "Delete all",
                    },
                    "descriptionStep2": "",
                    "identifierColumn": "UQ identifier",
                    "nameColumn": "Supervisor's name",
                    "reorderColumn": "Reorder items",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this item",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this item?",
                      "confirmationTitle": "Delete item",
                    },
                    "moveDownHint": "Move item down the order",
                    "moveUpHint": "Move item up the order",
                    "ordinalData": Array [
                      "First",
                      "Second",
                      "Third",
                      "Fourth",
                      "Fifth",
                      "Sixth",
                      "Seventh",
                      "Eighth",
                      "Ninth",
                      "Tenth",
                    ],
                    "selectHint": "Select this supervisor ([name]) to assign it as you",
                    "suffix": " listed supervisor",
                  },
                },
              }
            }
            name="supervisors"
            required={true}
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Field of research"
        >
          <WithStyles(Typography)>
            Select up to 3 Field of Research (FoR) codes at the 4 digit level
          </WithStyles(Typography)>
          <Field
            component={[Function]}
            disabled={false}
            distinctOnly={true}
            hideReorder={true}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "inputFieldHint": "Start typing code or field name and select from list",
                    "inputFieldLabel": "Field of research",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "deleteAll": "Remove all items",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all items?",
                      "confirmationTitle": "Delete all",
                    },
                    "nameColumn": "Field of research",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this item",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this item?",
                      "confirmationTitle": "Delete item",
                    },
                  },
                },
              }
            }
            maxCount={3}
            name="fieldOfResearch"
            required={true}
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Keywords"
        >
          <WithStyles(Typography)>
            Add up to 10 keywords that describe the content of the thesis
          </WithStyles(Typography)>
          <Field
            component={[Function]}
            disabled={false}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "addButtonLabel": "Add",
                    "inputFieldHint": "Type keywords",
                    "inputFieldLabel": "Keywords",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "deleteAll": "Remove all keywords",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all keywords?",
                      "confirmationTitle": "Delete all",
                    },
                    "nameColumn": "Keyword",
                    "reorderColumn": "Reorder keywords",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this keyword",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this keyword?",
                      "confirmationTitle": "Delete keyword",
                    },
                    "moveDownHint": "Move keyword down the order",
                    "moveUpHint": "Move keyword up the order",
                  },
                },
              }
            }
            maxCount={10}
            name="fez_record_search_key_keywords"
            required={true}
            searchKey={
              Object {
                "order": "rek_keywords_order",
                "value": "rek_keywords",
              }
            }
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Additional notes"
        >
          <Field
            component={[Function]}
            disabled={false}
            fullWidth={true}
            label="Notes (not publicly viewable)"
            multiline={true}
            name="comments"
            placeholder="Provide any additional information"
            rows={3}
            title="Additional notes"
            type="text"
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Upload files"
        >
          <Field
            component={[Function]}
            defaultQuickTemplateId={3}
            disabled={false}
            locale={
              Object {
                "accessTermsAndConditions": "I understand that the files indicated above as open access will be submitted as open access and will be made publicly available immediately or will be made available on the indicated embargo date.  All other files submitted will be accessible by UQ eSpace administrators.",
                "errorTitle": "Upload Errors",
                "fileUploadInstruction": <p>
                  Click here to select files, or drag files into this area to upload
                </p>,
                "fileUploadRestrictionHeading": "File upload restrictions",
                "fileUploadRestrictions": <div>
                  Maximum file size is 5Gb. 
                  <br />
                  PDF files must be saved using the following naming structure 
                  <b>
                    &lt;student number&gt;_&lt;degree type&gt;_&lt;document name&gt;.pdf
                  </b>
                  . Document name could be thesis, abstract, and etc. For example:
                  <ul>
                    <li>
                      s1234567_phd_thesis.pdf
                    </li>
                    <li>
                      s1234567_phd_abstract.pdf
                    </li>
                  </ul>
                  Supplementary audio files are to be in MP 3 format. 
                  <br />
                  Supplementary video files are to be in WMV or AVI format. 
                  <br />
                </div>,
                "instructions": "",
                "successMessage": "Successfully added [numberOfFiles] file(s) to upload queue.",
                "successTitle": "Success",
                "validation": Object {
                  "duplicateFiles": "File(s) ([fileNames]) are duplicates and have been ignored",
                  "invalidFileNames": "File(s) ([fileNames]) have invalid file name",
                  "notFiles": "Invalid files ([fileNames])",
                  "tooBigFiles": "File(s) ([fileNames]) exceed maximum allowed upload file size",
                  "tooManyFiles": "Maximum number of files ([maxNumberOfFiles]) has been exceeded. File(s) ([fileNames]) will not be uploaded",
                },
              }
            }
            name="files"
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      spacing={16}
    >
      <WithStyles(Grid)
        item={true}
        sm={true}
        xs={false}
      />
      <WithStyles(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(Button)
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="raised"
        >
          Cancel
        </WithStyles(Button)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(Button)
          color="primary"
          fullWidth={true}
          onClick={[Function]}
          variant="raised"
        >
          Deposit your thesis
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </form>
</StandardPage>
`;

exports[`SbsSubmission test should render sbs thesis submission form 2`] = `
<StandardPage
  title="Professional doctorate deposit"
>
  Hello
  <form>
    <NavigationDialogBox
      txt={
        Object {
          "cancelButtonLabel": "No",
          "confirmButtonLabel": "Yes",
          "confirmationMessage": "Are you sure you want to abandon workflow?",
          "confirmationTitle": "Abandon workflow",
        }
      }
      when={true}
    />
    <WithStyles(ConfirmDialogBox)
      locale={
        Object {
          "cancelButtonLabel": "No, continue editing",
          "confirmButtonLabel": "Yes, deposit thesis",
          "confirmationMessage": "You are about to deposit your thesis with attached files. Are you sure you want to proceed?",
          "confirmationTitle": "Thesis deposit",
        }
      }
      onAction={[MockFunction]}
      onRef={[Function]}
    />
    <WithStyles(Grid)
      container={true}
      spacing={24}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Thesis information"
        >
          <WithStyles(Grid)
            container={true}
            spacing={24}
          >
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Typography)
                variant="caption"
              >
                Thesis title
              </WithStyles(Typography)>
              <Field
                component={[Function]}
                disabled={false}
                height={50}
                name="thesisTitle"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                fullWidth={true}
                label="Author name"
                name="currentAuthor.0.nameAsPublished"
                placeholder=""
                required={true}
                rows={1}
                type="text"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                itemsList={
                  Array [
                    "MPhil Thesis",
                    "PhD Thesis",
                    "Professional Doctorate",
                  ]
                }
                locale={
                  Object {
                    "label": "Thesis type",
                  }
                }
                name="rek_genre_type"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                floatingLabelText="Institution name"
                hintText=""
                name="fez_record_search_key_org_name.rek_org_name"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                component={[Function]}
                disabled={false}
                floatingLabelText="Enrolling unit"
                hintText="Enrolling unit, eg. School of Business"
                name="fez_record_search_key_org_unit_name.rek_org_unit_name"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Field
                allowPartial={true}
                className="requiredHintField"
                component={[Function]}
                disabled={false}
                floatingTitle="Publication date"
                floatingTitleRequired={true}
                name="rek_date"
                required={true}
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Typography)
                variant="caption"
              >
                Abstract
              </WithStyles(Typography)>
              <Field
                component={[Function]}
                disabled={false}
                name="thesisAbstract"
                validate={
                  Array [
                    [Function],
                  ]
                }
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Supervisors"
        >
          <Field
            component={[Function]}
            disabled={false}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "addButton": "Add supervisor",
                    "descriptionStep1": "",
                    "descriptionStep1NoStep2": "List your current supervisors",
                    "identifierLabel": "UQ identifier (if available)",
                    "nameAsPublishedHint": "",
                    "nameAsPublishedLabel": "Enter each supervisorâ€™s name on a separate line e.g. Firstname Surname",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "contributorAssignmentColumn": "Select your name",
                    "deleteAll": "Remove all items",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all items?",
                      "confirmationTitle": "Delete all",
                    },
                    "descriptionStep2": "",
                    "identifierColumn": "UQ identifier",
                    "nameColumn": "Supervisor's name",
                    "reorderColumn": "Reorder items",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this item",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this item?",
                      "confirmationTitle": "Delete item",
                    },
                    "moveDownHint": "Move item down the order",
                    "moveUpHint": "Move item up the order",
                    "ordinalData": Array [
                      "First",
                      "Second",
                      "Third",
                      "Fourth",
                      "Fifth",
                      "Sixth",
                      "Seventh",
                      "Eighth",
                      "Ninth",
                      "Tenth",
                    ],
                    "selectHint": "Select this supervisor ([name]) to assign it as you",
                    "suffix": " listed supervisor",
                  },
                },
              }
            }
            name="supervisors"
            required={true}
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Field of research"
        >
          <WithStyles(Typography)>
            Select up to 3 Field of Research (FoR) codes at the 4 digit level
          </WithStyles(Typography)>
          <Field
            component={[Function]}
            disabled={false}
            distinctOnly={true}
            hideReorder={true}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "inputFieldHint": "Start typing code or field name and select from list",
                    "inputFieldLabel": "Field of research",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "deleteAll": "Remove all items",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all items?",
                      "confirmationTitle": "Delete all",
                    },
                    "nameColumn": "Field of research",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this item",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this item?",
                      "confirmationTitle": "Delete item",
                    },
                  },
                },
              }
            }
            maxCount={3}
            name="fieldOfResearch"
            required={true}
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Keywords"
        >
          <WithStyles(Typography)>
            Add up to 10 keywords that describe the content of the thesis
          </WithStyles(Typography)>
          <Field
            component={[Function]}
            disabled={false}
            locale={
              Object {
                "form": Object {
                  "locale": Object {
                    "addButtonLabel": "Add",
                    "inputFieldHint": "Type keywords",
                    "inputFieldLabel": "Keywords",
                  },
                },
                "header": Object {
                  "locale": Object {
                    "deleteAll": "Remove all keywords",
                    "deleteAllConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete all keywords?",
                      "confirmationTitle": "Delete all",
                    },
                    "nameColumn": "Keyword",
                    "reorderColumn": "Reorder keywords",
                  },
                },
                "row": Object {
                  "locale": Object {
                    "deleteHint": "Remove this keyword",
                    "deleteRecordConfirmation": Object {
                      "cancelButtonLabel": "No",
                      "confirmButtonLabel": "Yes",
                      "confirmationMessage": "Are you sure you want to delete this keyword?",
                      "confirmationTitle": "Delete keyword",
                    },
                    "moveDownHint": "Move keyword down the order",
                    "moveUpHint": "Move keyword up the order",
                  },
                },
              }
            }
            maxCount={10}
            name="fez_record_search_key_keywords"
            required={true}
            searchKey={
              Object {
                "order": "rek_keywords_order",
                "value": "rek_keywords",
              }
            }
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Additional notes"
        >
          <Field
            component={[Function]}
            disabled={false}
            fullWidth={true}
            label="Notes (not publicly viewable)"
            multiline={true}
            name="comments"
            placeholder="Provide any additional information"
            rows={3}
            title="Additional notes"
            type="text"
          />
        </StandardCard>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <StandardCard
          title="Upload files"
        >
          <Field
            component={[Function]}
            defaultQuickTemplateId={3}
            disabled={false}
            locale={
              Object {
                "accessTermsAndConditions": "I understand that the files indicated above as open access will be submitted as open access and will be made publicly available immediately or will be made available on the indicated embargo date.  All other files submitted will be accessible by UQ eSpace administrators.",
                "errorTitle": "Upload Errors",
                "fileUploadInstruction": <p>
                  Click here to select files, or drag files into this area to upload
                </p>,
                "fileUploadRestrictionHeading": "File upload restrictions",
                "fileUploadRestrictions": <div>
                  Maximum file size is 5Gb. 
                  <br />
                  PDF files must be saved using the following naming structure 
                  <b>
                    &lt;student number&gt;_&lt;degree type&gt;_&lt;document name&gt;.pdf
                  </b>
                  . Document name could be thesis, abstract, and etc. For example:
                  <ul>
                    <li>
                      s1234567_phd_thesis.pdf
                    </li>
                    <li>
                      s1234567_phd_abstract.pdf
                    </li>
                  </ul>
                  Supplementary audio files are to be in MP 3 format. 
                  <br />
                  Supplementary video files are to be in WMV or AVI format. 
                  <br />
                </div>,
                "instructions": "",
                "successMessage": "Successfully added [numberOfFiles] file(s) to upload queue.",
                "successTitle": "Success",
                "validation": Object {
                  "duplicateFiles": "File(s) ([fileNames]) are duplicates and have been ignored",
                  "invalidFileNames": "File(s) ([fileNames]) have invalid file name",
                  "notFiles": "Invalid files ([fileNames])",
                  "tooBigFiles": "File(s) ([fileNames]) exceed maximum allowed upload file size",
                  "tooManyFiles": "Maximum number of files ([maxNumberOfFiles]) has been exceeded. File(s) ([fileNames]) will not be uploaded",
                },
              }
            }
            name="files"
            validate={
              Array [
                [Function],
              ]
            }
          />
        </StandardCard>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      spacing={16}
    >
      <WithStyles(Grid)
        item={true}
        sm={true}
        xs={false}
      />
      <WithStyles(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(Button)
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="raised"
        >
          Cancel
        </WithStyles(Button)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(Button)
          color="primary"
          fullWidth={true}
          onClick={[Function]}
          variant="raised"
        >
          Deposit your thesis
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </form>
</StandardPage>
`;
